dll/win32/browseui/shellbrowser.cpp

```diff
@@ -3800,6 +3788,26 @@ LRESULT CShellBrowser::RelayCommands(UINT uMsg, WPARAM wParam, LPARAM lParam, BO
     return 0;
 }
 
+LRESULT CShellBrowser::OnBrowseUISettingChanged(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    /* Refresh child windows */
+    ::SendMessageW(fClientBars[BIInternetToolbar].hwnd, uMsg, wParam, lParam);
+
+    /* Refresh status bar */
+    if (fStatusBar)
+    {
+        ::ShowWindow(fStatusBar, m_settings.fStatusBarVisible ? SW_SHOW : SW_HIDE);
+        RepositionBars();
+    }
+
+    return 0;
+}
+
+LRESULT CShellBrowser::OnGetBrowseUISettings(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    return (LRESULT)&m_settings;
```

Comment:
This is bad usage of LRESULT. (See my previous comment.)
You should instead implement this handler so that it returns the pointer via e.g. the LPARAM.
```suggestion
    if (!lParam)
        return ERROR_INVALID_PARAMETER;

    *(BrowseUISettings**)lParam = &m_settings;
    return NO_ERROR;
```